VideoLocker/src/main/java/org/edx/mobile/view/UserProfileFragment.java

```diff
@@ -0,0 +1,246 @@
+
+import roboguice.fragment.RoboFragment;
+import roboguice.inject.InjectExtra;
+
+public class UserProfileFragment extends RoboFragment {
+
+    @InjectExtra(UserProfileActivity.EXTRA_USERNAME)
+    private String username;
+
+    @Nullable
+    private GetAccountTask getAccountTask;
+
+    @Nullable
+    private Account account;
+
+    @Nullable
+    private ViewHolder viewHolder;
+
+    @Nullable
+    MenuItem editProfileMenuItem;
+
+    @Inject
+    private Router router;
+
+    private boolean isViewingOwnProfile;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setRetainInstance(true);
+        setHasOptionsMenu(true);
+
+        final ProfileModel model = new PrefManager(getActivity(), PrefManager.Pref.LOGIN).getCurrentUserProfile();
+        isViewingOwnProfile = null != model && model.username.equalsIgnoreCase(username);
+
+        getAccountTask = new GetAccountTask(getActivity(), username) {
+            @Override
+            protected void onSuccess(Account account) throws Exception {
+                UserProfileFragment.this.account = account;
+                if (null != viewHolder) {
+                    viewHolder.setAccount(account);
+                }
+            }
+        };
+        getAccountTask.setTaskProcessCallback(null); // Disable default loading indicator, we have our own
+        getAccountTask.execute();
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+        if (isViewingOwnProfile) {
+            inflater.inflate(R.menu.edit_profile, menu);
+            editProfileMenuItem = menu.findItem(R.id.edit_profile).setIcon(
+                    new IconDrawable(getActivity(), Iconify.IconValue.fa_pencil)
+                            .actionBarSize().colorRes(R.color.edx_white));
+        }
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == R.id.edit_profile) {
+            router.showUserProfileEditor(getActivity(), username);
+            return true;
+        } else {
+            return super.onOptionsItemSelected(item);
+        }
+    }
+
+    @Override
+    public void onDestroyOptionsMenu() {
+        super.onDestroyOptionsMenu();
+        editProfileMenuItem = null;
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_user_profile, container, false);
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        viewHolder = new ViewHolder(view, isViewingOwnProfile, new Runnable() {
+            @Override
+            public void run() {
+                router.showUserProfileEditor(getActivity(), username);
+            }
+        });
```

Comment:
Hmm, if you make the `ViewHolder` a non-static class, then it could have access to all these fields directly (and shouldn't need to be passed this click handling `Runnable`).